prompting/llms/utils.py

```diff
@@ -0,0 +1,33 @@
+import re
+import torch
+import bittensor as bt
+
+def contains_gpu_index_in_device(device: str) -> bool:
```

Comment:
what does this do? check if cuda:i is available?

Reply:
This extracts if there any particular gpu index assigned to the device, such as `cuda:3`. This impacts on how we later calculate available space in the device as we need an specific index to call `torch.cuda.mem_get_info()`.